### PR TITLE
NSPC: Add support for namespaces

### PR DESCRIPTION
**Pull request by mbtools**
Part 2 of #835

If a repo contains any namespaced object, then this PR will make namespaces part of the repository. So far, serializing and updating an existing namespace is supported.

Note: Only the repair key will be serialized. The developer key for the namespace is *not* included.

abapGit can not (yet) create the namespace automatically. This will require a bit of refactoring which will then also support creating the top repo package automatically. Coming soon. 

See readme of test case https://github.com/abapGit-tests/NSPC

Example of repo with namespace (NSPC):

image

### CODE REVIEW

File: src/objects/core/zcl_abapgit_tadir.clas.abap [hunk 1]
```diff
@@ -112,6 +116,31 @@ CLASS zcl_abapgit_tadir IMPLEMENTATION.
       ENDIF.
     ENDLOOP.
 
+    LOOP AT rt_tadir ASSIGNING <ls_tadir> WHERE obj_name(1) = '/'.
```
Comment (larshp) [inline src/objects/core/zcl_abapgit_tadir.clas.abap:119]: suggest extracting logic to new private method

Comment (mbtools) [reply src/objects/core/zcl_abapgit_tadir.clas.abap]: I split `build` into smaller methods. the diff looks crazy but it's just refactoring.

**Comment by larshp** [comment]
whats the plan for handling the top package?

**Comment by mbtools** [comment]
Top package auto-create: https://github.com/abapGit/abapGit/pull/4431

**Pull request merged**